q2_diversity_lib/tests/test_beta.py

```diff
@@ -321,10 +322,152 @@ def test_accepted_types_have_consistent_behavior(self):
+            for id2 in actual.ids:
+                npt.assert_almost_equal(actual[id1, id2], expected[id1, id2])
+
+    def test_canberra_adkins(self):
+        t = biom.Table(np.array([[0, 0], [0, 1], [1, 2]]),
+                       ['O1', 'O2', 'O3'],
+                       ['S1', 'S2'])
+        t = Artifact.import_data('FeatureTable[Frequency]', t)
+        d = (1. / 2.) * sum([abs(0. - 1.) / (0. + 1.),
+                             abs(1. - 2.) / (1. + 2.)])
+        expected = skbio.DistanceMatrix(np.array([[0.0, d], [d, 0.0]]),
+                                        ids=['S1', 'S2'])
+        actual = self.method(table=t, metric='canberra_adkins')
+        actual = actual[0].view(skbio.DistanceMatrix)
+
+        self.assertEqual(actual.ids, expected.ids)
+        for id1 in actual.ids:
+            for id2 in actual.ids:
+                npt.assert_almost_equal(actual[id1, id2], expected[id1, id2])
+
+    def test_beta_canberra_adkins_negative_values(self):
+        t = biom.Table(np.array([[0, 0], [0, 1], [-1, -2]]),
+                       ['O1', 'O2', 'O3'],
+                       ['S1', 'S2'])
+        t = Artifact.import_data('FeatureTable[Frequency]', t)
+
+        with self.assertRaisesRegex(ValueError, 'cannot.*negative values'):
+            self.method(table=t, metric='canberra_adkins')
+
+    def test_jensenshannon(self):
+        actual = self.method(table=self.table, metric='jensenshannon')
+        # expected computed with scipy.spatial.distance.jensenshannon
+        expected = skbio.DistanceMatrix([[0.0000000, 0.4645014, 0.52379239],
+                                         [0.4645014, 0.0000000, 0.07112939],
+                                         [0.52379239, 0.07112939, 0.0000000]],
+                                        ids=['S1', 'S2', 'S3'])
+
+        actual = actual[0].view(skbio.DistanceMatrix)
+        self.assertEqual(actual.ids, expected.ids)
+        for id1 in actual.ids:
+            for id2 in actual.ids:
+                npt.assert_almost_equal(actual[id1, id2], expected[id1, id2])
+
+
+class BetaPhylogeneticPassthroughTests(TestPluginBase):
+    package = 'q2_diversity_lib.tests'
+
+    def setUp(self):
+        super().setUp()
+        self.method = self.plugin.actions['beta_phylogenetic_passthrough']
+        self.available_metrics = beta.METRICS['PHYLO']['UNIMPL']
+        empty_table = biom.Table(np.array([]), [], [])
+        self.empty_table = Artifact.import_data('FeatureTable[Frequency]',
+                                                empty_table)
+        crawford_tbl = self.get_data_path('crawford.biom')
+        self.crawford_tbl = Artifact.import_data('FeatureTable[Frequency]',
+                                                 crawford_tbl)
+        crawford_tree = self.get_data_path('crawford.nwk')
+        self.crawford_tree = Artifact.import_data('Phylogeny[Rooted]',
+                                                  crawford_tree)
+
+    def testMethod(self):
```

Comment:
Should this be `test_method`?

```suggestion
    def test_method(self):
```